packages/react/src/components/Details/Details.tsx

```diff
@@ -3,26 +3,35 @@ import cl from 'clsx/lite';
 import type { HTMLAttributes, ReactNode } from 'react';
 import { forwardRef, useEffect, useRef } from 'react';
 import '@u-elements/u-details';
+import type { DefaultProps } from '../../types';
+import type { MergeRight } from '../../utilities';
 
-export type DetailsProps = {
-  /**
-   * Controls open-state.
-   *
-   * Using this removes automatic control of open-state
-   *
-   * @default undefined
-   */
-  open?: boolean;
-  /**
-   * Defaults the details to open if not controlled
-   * @default false
-   */
-  defaultOpen?: boolean;
-  /** Callback function when Details toggles due to click on summary or find in page-search */
-  onToggle?: (event: Event) => void;
-  /** Content should be one `<Details.Summary>` and `<Details.Content>` */
-  children?: ReactNode;
-} & Omit<HTMLAttributes<HTMLDetailsElement>, 'onToggle'> &
+export type DetailsProps = MergeRight<
+  DefaultProps & HTMLAttributes<HTMLDetailsElement>,
+  {
+    /**
+     * Specify which color palette to use.
+     */
+    'data-color'?: DefaultProps['data-color'];
```

Comment:
`DefaultProps` comes wiht `data-color`, do we need to specify it again?

Reply:
Something with `MergeRight` or how react-docgen parses types not working as intended?

Reply:
The only reason I'm repeating it, is to make the documentation make sense. The `DefaultProps['data-color'] doc comment says (emphasis mine):

>  Sets a color palette which may be used by **descendants**. **Does not affect this component.** If left unspecified, the color is inherited from the nearest ancestor with data-color.

When I implemented the new color system I landed on this as the best default documentation, since it would be confusing for components that don't change color based on this prop to have "Specify which color palette to use.". But if you have any ideas or suggestions on how to do this more clearly, I'm all ears 😊 

**However**, looking at Details' implementation, both the type and doc comment should actually be different. It should be something like this:
```tsx
    /**
     * Specify which color palette to use.
     * If left unspecified, the color is inherited from the nearest ancestor with data-color.
     */
    'data-color'?: 'subtle' | Color;
```